src/App.js

```diff
@@ -1,25 +1,80 @@
-import logo from './logo.svg';
-import './App.css';
-
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+import React, { useState, useEffect } from 'react'
+import './App.css'
+import Header from './functionBased/components/Header'
+import { v4 as uuid } from 'uuid'
+import TodoList from './functionBased/components/TodoList'
+
+const App = () => {
+	const [todos, setTodos] = useState(getInitialTodos())
+
+	const handleChange = id => {
+		setTodos(prevState => {
+			prevState.map(todo => {
+				if (todo.id === id) {
+					return {
+						...todo,
+						completed: !completed,
+					}
+				}
+				return todo
+			})
+		})
+	}
+
+	const addTodoItem = title => {
+		const newTodo = {
+			id: uuid(),
+			title: title,
+			completed: false,
+		}
+		setTodos([...todos, newTodo])
+	}
+
+	const delTodo = id => {
+		setTodos([...todos.filter(todo => todo.id !== id)])
+	}
+
+	const setUpdate = (updatedTitle, id) => {
+		setTodos(
+			todos.map(todo => {
+				if (todo.id === id) {
+					todo.title = updatedTitle
+				}
+				return todo
+			}),
+		)
+	}
+
+	function getInitialTodos() {
+		const savedTodos = JSON.parse(localStorage.getItem('todos'))
+		return savedTodos || []
+	}
+
+	useEffect(() => {
+		localStorage.setItem('todos', JSON.stringify(todos))
+	}, [todos])
+
+	return (
+		<div className='container'>
+			<div className='inner'>
+				<Header />
+				<InputTodo addTodoProps={addTodoItem} />
+				<TodoList
+					todos={todos}
+					handlePropsChange={handleChange}
+					deleteTodoProps={delTodo}
+					setUpdate={setUpdate}
+				/>
+			</div>
+		</div>
+	)
 }
 
-export default App;
+export default App
+
+// 	componentWillUnmount() {
+// 		console.log('Cleaning up...')
+// 	}
+// }
+
+// export default App
```

Comment:
- [ ] I will strongly recommend you remove all commented code from your codebase so that your codebase can be clean, readable and easy to maintain 😄